### PR TITLE
NGG coding rework: Simplify some member initialization

### PR DESCRIPTION
**Pull request by amdrexu**
Initialize members in the class declaration and remove the useless member m_hasTcs.

### CODE REVIEW

**Comment by amdvlk-admin** [comment]
### Test summary for commit 31b5c003448c73dae0070ac991f3200b8ba26abf
<details><summary>CTS tests (Failed: 1/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38863/70485 (55.1%)</li>
<li><details><summary>Failed: 1/70485 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.api.buffer_marker.compute.default_mem.bottom_of_pipe.memory_dep.dispatch
Stack trace: Flake
```
</p></details></li>

<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39114/70485 (55.5%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/NggPrimShader.cpp [hunk 1]
```diff
@@ -79,7 +79,7 @@ enum {
 NggPrimShader::NggPrimShader(PipelineState *pipelineState)
     : m_pipelineState(pipelineState), m_context(&pipelineState->getContext()),
       m_gfxIp(pipelineState->getTargetInfo().getGfxIpVersion()), m_nggControl(m_pipelineState->getNggControl()),
-      m_ldsManager(nullptr), m_constPositionZ(false), m_builder(new IRBuilder<>(*m_context)) {
+      m_builder(std::make_unique<IRBuilder<>>(pipelineState->getContext())) {
```
Comment (nhaehnle) [inline lgc/patch/NggPrimShader.cpp:82]: While you're at it, why is `m_builder` a unique pointer in the first place? Couldn't it be a plain `IRBuilder<>` object and avoid the heap allocation?

On second thought, this adds a whole lot of additional churn, but if you could do it anyway (maybe in a separate change) I'd appreciate it.

**Review by nhaehnle — APPROVED** [review]
I have a comment inline, but I think this is fine to go in as-is, and it is indeed a nice cleanup!

File: lgc/patch/NggPrimShader.cpp (hunk 1 above)
Comment (amdrexu) [reply lgc/patch/NggPrimShader.cpp:82]: Thank you very much. I will try to correct it in a separate change. I plan to do a series of changes to refine NGG coding in these weeks to get rid of unclear coding issues and vague design logic.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 31b5c003448c73dae0070ac991f3200b8ba26abf
<details><summary>CTS tests (Failed: 0/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38864/70485 (55.1%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39114/70485 (55.5%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 8.6, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 6ad8c28d5816d069f02417be189685225bffb20d
<details><summary>CTS tests (Failed: 0/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38864/70485 (55.1%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39114/70485 (55.5%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

File: lgc/patch/NggPrimShader.cpp (hunk 1 above)
Comment (amdrexu) [reply lgc/patch/NggPrimShader.cpp]: I made an extra change to correct the use of IR builder.

**Comment by amdvlk-admin** [comment]
### Test summary for commit 3bed03675ed7050c2329819209314814439f5096
<details><summary>CTS tests (Failed: 0/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38864/70485 (55.1%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39114/70485 (55.5%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

**Comment by amdvlk-admin** [comment]
### Test summary for commit 98ad5553a7682cc751aa3bd96706534604fc03e0
<details><summary>CTS tests (Failed: 2/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38864/70485 (55.1%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39112/70485 (55.5%)</li>
<li><details><summary>Failed: 2/70485 (0.0%)</summary><p>


Failures:
```
FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_8.triangles.degenerate.max
Stack trace: Fail

FAILURE: dEQP-VK.rasterization.conservative.overestimate.samples_8.triangles.degenerate.min
Stack trace: Fail
```
</p></details></li>

<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

**Review by nhaehnle — APPROVED** [review]
Thanks, LGTM

**Comment by amdvlk-admin** [comment]
### Test summary for commit 98ad5553a7682cc751aa3bd96706534604fc03e0
<details><summary>CTS tests (Failed: 0/277783)</summary><ul><li>Built with version 1.3.0.0</li>
<details open><summary>Ubuntu 22.04, Navi3x</summary><ul>

<li>Passed: 38864/70485 (55.1%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31621/70485 (44.9%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 22.04, Navi2x</summary><ul>

<li>Passed: 39114/70485 (55.5%)</li>
<li>Failed: 0/70485 (0.0%)</li>
<li>Not Supported: 31371/70485 (44.5%)</li><li>Warnings: 0/70485 (0.0%)</li>
</ul></details>

<details open><summary>Ubuntu 20.04, Gfx10</summary><ul>

<li>Passed: 35959/70487 (51.0%)</li>
<li>Failed: 0/70487 (0.0%)</li>
<li>Not Supported: 34528/70487 (49.0%)</li><li>Warnings: 0/70487 (0.0%)</li>
</ul></details>

<details open><summary>Rhel 9.0, Gfx9</summary><ul>

<li>Passed: 33678/66326 (50.8%)</li>
<li>Failed: 0/66326 (0.0%)</li>
<li>Not Supported: 32648/66326 (49.2%)</li><li>Warnings: 0/66326 (0.0%)</li>
</ul></details>


</ul></details>

**Pull request merged**